### PR TITLE
Add separate helm values for the liveness and readiness probes

### PR DESCRIPTION
**Pull request by tobiasgiese**
This PR adds separate helm values for the probes. This is necessary to override the health port in the probes.
Follow-up to https://github.com/kubernetes-sigs/node-feature-discovery/pull/1885

Also the `helm.md` has been reformatted.

The updated helm documentation can be seen here: https://deploy-preview-1913--kubernetes-sigs-nfd.netlify.app/deployment/helm

### CODE REVIEW

File: deployment/helm/node-feature-discovery/templates/worker.yaml [hunk 1]
```diff
@@ -47,9 +47,38 @@ spec:
...
+        {{- with .Values.worker.readinessProbe.initialDelaySeconds }}
+          initialDelaySeconds: {{ . }}
+        {{- end }}
+        {{- with .Values.worker.readinessProbe.failureThreshold }}
+          failureThreshold: {{ . }}
+        {{- end }}
+        {{- with .Values.worker.readinessProbe.periodSeconds }}
+          periodSeconds: {{ . }}
+        {{- end }}
+        {{- with .Values.worker.readinessProbe.timeoutSeconds }}
```
Comment (adrianchiris) [inline deployment/helm/node-feature-discovery/templates/worker.yaml:76]: this works when timeoutSecond is not provided in values? just making sure :)

**Review by adrianchiris — COMMENTED** [review]
overall lgtm

File: deployment/helm/node-feature-discovery/templates/worker.yaml (hunk 1 above)
Comment (tobiasgiese) [reply deployment/helm/node-feature-discovery/templates/worker.yaml:76]: it works, yes. Currently there are only comments in the values.yaml :)

Comment (ArangoGutierrez) [reply deployment/helm/node-feature-discovery/templates/worker.yaml:76]: Given the NV-heavy presence on this PR, I'll leave the final decision to @marquiz

**Review by marquiz — APPROVED** [review]
Thanks @tobiasgiese for the improvement.
/lgtm

I have some changes cooking up on this front, too, so stay tuned

**Pull request merged**